scheduler/services/resource/ocs_resource_service.py

```diff
@@ -0,0 +1,568 @@
+# Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+# For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+import csv
+import logging
+import os
+from copy import copy
+from datetime import datetime, timedelta
+from typing import Dict, Final, List, NoReturn, Set, Tuple
+
+import gelidum
+import requests
+from lucupy.helpers import str_to_bool
+from lucupy.minimodel import ALL_SITES
+from openpyxl import load_workbook
+
+from definitions import ROOT_DIR
+from .filters import *
+from .google_drive_downloader import GoogleDriveDownloader
+from .night_resource_configuration import *
+from .ocs_resource_service_exceptions import *
+from scheduler.core.meta import Singleton
+from scheduler.core.resourcemanager import ResourceManager
+
+
+@final
+class OcsResourceService(ResourceManager, metaclass=Singleton):
+    """
+    This is a mock for the future Resource service, used for OCS.
+    It reads data regarding availability of instruments, IFUs, FPUs, MOS masks, etc. at each Site for given dates.
+
+    It can then be queried to receive a set of Resource (usually with barcode IDs, except for instruments) for a
+    given site on a given night.
+
+    It caches and reuses Resources by ID as best as possible to minimize the number of Resource objects existing
+    at any given time. Since Resource is immutable, this should be fine.
+
+    Note that this is a Singleton class, so new instances do not need to be created.
+    """
+    # The Google ID of the telescope configuration file.
+    _SITE_CONFIG_GOOGLE_ID: Final[str] = '1QRalQNEaX-bcyrPG6mfKnv01JVMaGHwy'
```

Comment:
There is not issue with this being public?

Reply:
I don't think so. We had the actual spreadsheet itself checked into the repo before... now we just have a link to it instead on the Google drive. I was thinking at first that it should be perhaps kept a secret, but I doubt old telescope configs are secret. We can confirm with Bryan and Kristin on Tuesday.

Reply:
Great, if is per document than should be not issue.